data-pipeline-flink/user-cache-updater-2.0/src/main/scala/org/sunbird/dp/usercache/util/UserMetadataUpdater.scala

```diff
@@ -89,10 +89,19 @@ object UserMetadataUpdater {
       }
 
       //Flatten User Type and subType
-      val profileUserType = response.profileUserType
-      if (null != profileUserType && !profileUserType.isEmpty) {
-        userCacheData.+=(config.userTypeKey -> profileUserType.getOrDefault(config.`type`, ""),
-          config.userSubtypeKey -> profileUserType.getOrDefault(config.subtype, ""))
+      val profileUserTypes = response.profileUserTypes
+      if (null != profileUserTypes && !profileUserTypes.isEmpty) {
+        var userTypeValue = Array[String]()
+        var userSubtypeValue = Array[String]()
+        profileUserTypes.forEach(userType => {
+          val typeVal:String = userType.get(config.`type`)
+          val subTypeVal:String = userType.get(config.subtype)
+
+          if (typeVal != null && typeVal.nonEmpty && !userTypeValue.contains(typeVal)) userTypeValue :+= typeVal
+          if (subTypeVal != null && subTypeVal.nonEmpty && !userSubtypeValue.contains(subTypeVal)) userSubtypeValue :+= subTypeVal
+        })
```

Comment:
Why don't you create a function to fetch the type and subtype value array? Don't use `var`.

Reply:
Since getRegisteredUserInfo has all the functions regarding user cache map formation, I have added usertype string formation also in the same function. Now I have added a new function to create the user type strings.